aws/sagemaker/large-model-inference/sample-llm/multi_lora_adapter_inference_advanced.ipynb

```diff
@@ -0,0 +1,490 @@
+   "metadata": {},
+   "source": [
+    "We will be deploying an endpoint with 2 LoRA adapters. These are the models we will be using:\n",
+    "- Base Model: https://huggingface.co/huggyllama/llama-7b\n",
+    "- LoRA Fine Tuned Adapter 1: https://huggingface.co/tloen/alpaca-lora-7b\n",
+    "- LoRA Fine Tuned Adapter 2: https://huggingface.co/22h/cabrita-lora-v0-1"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8d57e9d5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!rm -rf lora-multi-adapter\n",
+    "!mkdir -p lora-multi-adapter/adapters"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "978224f5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "snapshot_download(\"tloen/alpaca-lora-7b\", local_dir=\"lora-multi-adapter/adapters/eng_alpaca\", local_dir_use_symlinks=False)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ba747f00",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "snapshot_download(\"22h/cabrita-lora-v0-1\", local_dir=\"lora-multi-adapter/adapters/portuguese_alpaca\", local_dir_use_symlinks=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d89777de",
+   "metadata": {},
+   "source": [
+    "## Creating Inference Handler and DJL Serving Configuration"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "89bd1577",
+   "metadata": {},
+   "source": [
+    "The following files cover the model server configuration (`serving.properties`) and custom inference handler (`model.py`). The custom inference handler is optional and if not specified, default handler from djl-serving will be used. This configuration can be used as an example to write your own inference handler for different models. \n",
+    "\n",
+    "The core structure to cover here is the model directory. We include both the base model and LoRA adapters in the model directory like this:\n",
+    "\n",
+    "```\n",
+    "|- model_dir\n",
+    "    |- adapters/\n",
+    "        |--- <adapter_1>/\n",
+    "        |--- <adapter_2>/\n",
+    "        |--- ...\n",
+    "        |--- <adapter_n>/\n",
+    "    |- serving.properties\n",
+    "    |- model.py (optional)\n",
+    "\n",
+    "```\n",
+    "\n",
+    "Each of the adapters in the `adapters` directory contains the LoRA adapter artifacts. Typically there are two files: `adapter_model.bin` and `adapter_config.json` which are the adapter weights and adapter configuration respectively. These are typically obtained from the Peft library via the `PeftModel.save_pretrained()` method."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5e988852",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%writefile lora-multi-adapter/serving.properties\n",
+    "engine=Python\n",
+    "option.model_id=huggyllama/llama-7b\n",
+    "option.dtype=fp16\n",
+    "option.entryPoint=model.py\n",
+    "option.tensor_parallel_degree=1\n",
```

Comment:
Should we show rolling batch as advanced example?

Reply:
Wouldn't that not work with a custom handler